test/InputStates-mocha.js

```diff
@@ -0,0 +1,261 @@
+/**
+ * File: InputStatess-mocha.js
+ */
+
+var chai = require('chai');
+var should = chai.should();
+var expect = chai.expect;
+
+var InputStates = require('../components/InputStates');
+var componentFactory = require('../components/event-component-factory.js');
+var test = require('./common-test');
+
+describe("InputStates", function() {
+
+    it("should exist as a function", function() {
+        InputStates.should.exist;
+        InputStates.should.be.a('function');
+    });
+
+    it("should throw an error if no port specified", function() {
+        expect(InputStates).to.throw(Error, /Invalid arguments specified for InputState/);
+    });
+
+    it("should be undefined if no InputState has been set for a port", function() {
+       expect( InputStates('input') ).to.be.undefined;
+    });
+
+    it("should get and set state for an IIP port", function() {
+
+        // Set test input data
+        var testData = "Test State";
+        var testLm = "LM1328113669.00000000000000001";
+        var initialTestState = { data: testData, 
+                                 lm: testLm };
+        var inportName = "input";
+
+        // Set the input state for an "input" port
+        var context = InputStates( inportName, 
+                                   initialTestState );
+
+        // Context here will be the node.js context since mocha is running on that by default
```

Comment:
This code is not dealing with NoFlo directly.  It will return whatever context it was called with.  If you call it from a VNI and pass the VNI context, it will return the VNI context.  If you call it with NodeJS context, (and I have tests that do), it will return that context.  This code has no noflo dependencies.